src/api/goods.ts

```
@@ -4,13 +4,33 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+
+      return response.json();
+    })
+    .catch(() => {
+      throw new Error('Failed to fetch goods');
+    });
```

This `.catch` block hides the original error. For example, the specific error thrown on line 10 with the HTTP status will be replaced by the generic 'Failed to fetch goods' message here, making debugging harder. It would be better to let the original, more detailed error propagate to the caller. This same issue is present in the `.catch` blocks for `get5First` and `getRedGoods`.